sale_import_base/components/importer_sale_channel.py

```diff
@@ -0,0 +1,339 @@
+#  Copyright (c) Akretion 2020
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from copy import deepcopy
+
+from marshmallow_objects import ValidationError as MarshmallowValidationError
+
+from odoo.exceptions import ValidationError
+
+from odoo.addons.component.core import Component
+
+MAPPINGS_SALE_ORDER_ADDRESS_SIMPLE = [
+    "name",
+    "street",
+    "street2",
+    "zip",
+    "city",
+    "email",
+]
+
+
+class ImporterSaleChannel(Component):
+    _inherit = "processor"
+    _name = "importer.sale.channel"
+    _apply_on = ["sale.order"]
+    _usage = "json_import"
+
+    def run(self, raw_data):
+        """
+        :param raw_data: json-like string
+        :return: generated sale order
+        """
+        # we need this step because the chunk data is stored as
+        # a string (so we can edit it)
+        try:
+            so_datamodel_load = self.env.datamodels["sale.order"].load_json(raw_data)
+        except MarshmallowValidationError:
+            raise ValidationError(MarshmallowValidationError)
+        except Exception:
+            raise
+        json_data_initial = so_datamodel_load.dump()
+        json_data = deepcopy(json_data_initial)
```

Comment:
I think we can remove the deepcopy (or found a solution to remove it)

Reply:
I think previously it was in tests, and the solution was to make an @property on the TestCase to return a deepcopy.
The deepcopy happens because during processing, data is transformed in-place (instead of having something like 

data_step2 = function1(data_step1)
data_step3 = function2(data_step2) 
(...)

So we need one for the processing and one for the finalize function (which looks at the original data)

How should we act on this?

Reply:
solved by refactor